File: examples/autogen/example_1/PYTHON_AUTOGEN/example_1.py
from .attributes.json_attr import json_serializable_class_attribute
from .attributes.json_attr import json_deserializable_class_attribute
from .attributes.category_attr import category_class_attribute
from .abstract_class.abstract_class import AbstractPersonClass


@json_serializable_class_attribute
@json_deserializable_class_attribute
@category_class_attribute(name='staff')
class Person(object):

    __class_name = "Person"
    __ten = 10
    __twelve = None

    def __init__(self):
        self.__abstract_class = AbstractPersonClass()

        self.__is_man = None
        self.__age = None
        self.__first_name = None
        self.__last_name = None
        self.__thirty = 30
        self.__money_is_not_a_problem = None

    @property
    def is_man(self):
        return self.__is_man

    @is_man.setter
    def is_man(self, value):
        self.__is_man = value

    @property
    def age(self):
        return self.__age

    @age.setter
    def age(self, value):
        self.__age = value

    @property
    def first_name(self):
        return self.__first_name

    @first_name.setter
    def first_name(self, value):
        self.__first_name = value

    @property
    def last_name(self):
        return self.__last_name

    @last_name.setter
    def last_name(self, value):
        self.__last_name = value

    @property
    def thirty(self):
        return self.__thirty

    @property
    def money_is_not_a_problem(self):
        return self.__money_is_not_a_problem

    @money_is_not_a_problem.setter
    def money_is_not_a_problem(self, value):
        self.__money_is_not_a_problem = value

    @property
    def class_name(self):
        return type(self).__class_name

    @property
    def ten(self):
        return type(self).__ten

    @property
    def twelve(self):
        return type(self).__twelve

    @twelve.setter
    def twelve(self, value):
        type(self).__twelve = value

    @property
    def _abstract_class(self) -> AbstractPersonClass:
        return self.__abstract_class

    def get_name(self) -> str:
        pass

File: examples/autogen/example_1/JAVA_AUTOGEN/example_1.java
import java.io.File;
import java.io.IOException;
import com.fasterxml.jackson.databind.ObjectMapper;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({ "thirty", "moneyIsNotAProblem", "className", "ten" })
public class Person {
    public static final String className = "Person";
    public static final Integer ten = 10;
    public static Integer twelve = null;

    public Boolean isMan = null;

    public Integer age = null;

    public String firstName = null;

    public String lastName = null;

    public final Integer thirty = 30;
    public Boolean moneyIsNotAProblem = null;

    @JsonIgnore
    private final ObjectMapper m_mapper = new ObjectMapper();

    public Boolean getIsMan() {
        return this.isMan;
    }

    public void setIsMan(Boolean isMan) {
        this.isMan = isMan;
    }

    public Integer getAge() {
        return this.age;
    }

    public void setAge(Integer age) {
        this.age = age;
    }

    public String getFirstName() {
        return this.firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return this.lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public Integer getThirty() {
        return this.thirty;
    }

    public Boolean getMoneyIsNotAProblem() {
        return this.moneyIsNotAProblem;
    }

    public void setMoneyIsNotAProblem(Boolean moneyIsNotAProblem) {
        this.moneyIsNotAProblem = moneyIsNotAProblem;
    }

    public String toJsonString() throws IOException {
        return this.m_mapper.writerWithDefaultPrettyPrinter().writeValueAsString(this);
    }

    public void saveToJson(String pathToSaveTo) throws IOException {
        this.m_mapper.writerWithDefaultPrettyPrinter().writeValue(new File(pathToSaveTo), this);
    }

    public void fromJsonString(String jsonString) throws IOException {
        Person loaded = this.m_mapper.readValue(jsonString, Person.class);
        this.isMan = loaded.isMan;
        this.age = loaded.age;
        this.firstName = loaded.firstName;
        this.lastName = loaded.lastName;
        
    }

    public void loadFromJson(String pathToLoadFrom) throws IOException {
        Person loaded = this.m_mapper.readValue(new File(pathToLoadFrom), Person.class);
        this.isMan = loaded.isMan;
        this.age = loaded.age;
        this.firstName = loaded.firstName;
        this.lastName = loaded.lastName;
        
    }

    @JsonIgnore
    public String getCategory() {
        return "staff";
    }

    @JsonIgnore
    public String getName() {
        throw new UnsupportedOperationException();
    }
}